catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/filter/FilterReader.java

```diff
@@ -35,14 +50,38 @@ public FilterReader() throws JAXBException {
   @SuppressWarnings("unchecked")
   private <T> JAXBElement<T> unmarshal(InputStream inputStream, Class<T> tClass)
       throws JAXBException {
+    SAXParser parser;
+    try {
+      parser = factory.newSAXParser();
+    } catch (SAXException | ParserConfigurationException e) {
+      throw new JAXBException("Could not create SAX parser", e);
+    }
+
+    XMLReader reader;
+    try {
```

Comment:
✏️ I'm not sure I see the advantage in separating the two try/catch blocks. Is there really value in distinguishing between a failure to create a parser and failure to get a reader to the calling code?